src/blueapi/preprocessors/attach_metadata.py

```diff
@@ -41,32 +42,10 @@ def attach_metadata(
     Yields:
         Iterator[Msg]: Plan messages
     """
-
-    group_in_progress = False
-
-    for message in plan:
-        # If the first stage in a series of stages is detected,
-        # update the directory provider and create a new group.
-        if (message.command == "stage") and (not group_in_progress):
-            yield from bps.wait_for([provider.update])
-            group_in_progress = True
-        # Mark if detectors are being unstaged so that the start
-        # of the next sequence of stages is detectable.
-        elif message.command == "unstage":
-            group_in_progress = False
-
-        # If a run is being opened, attempt to bundle the information
-        # on any existing group into the start document.
-        if message.command == "open_run":
-            # Handle the case where we're opening a run but no detectors
-            # have been staged yet. Common for nested runs.
-            if not group_in_progress:
-                yield from bps.wait_for([provider.update])
-            directory_info = provider()
-            message.kwargs[DATA_SESSION] = directory_info.filename_prefix
-
-        # This is a preprocessor so we yield the original message.
-        yield message
+    directory_info = provider()
```

Comment:
As a result of no longer using the `update()` method, the type annotation of provider in the function signature can be changed to `DirectoryProvider`